benchmarking-cli/src/main.rs

```diff
@@ -0,0 +1,157 @@
+        .unwrap_or(&vec![])
+        .iter()
+        .map(|v| v.parse::<u64>().unwrap())
+        .collect::<Vec<u64>>()
+}
+
+#[derive(Parser)]
+#[clap(
+    author = "Polygon Miden",
+    version,
+    about = "A very simple benchmarking CLI for Miden examples"
+)]
+
+struct Cli {
+    #[arg(
+        short,
+        long,
+        help("Provide example name as in ../examples"),
+        required(true)
+    )]
+    example: String,
+
+    #[arg(
+        short,
+        long,
+        help("Set to 'high' if 128-bit is needed"),
+        default_value("")
+    )]
+    security: String,
+
+    #[arg(
+        short,
+        long,
+        help("Set the number of desired stack outputs"),
+        default_value("1")
+    )]
+    output: usize,
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    // The CLI should take a path to the example directory as input, execute/prove the example, and output the following metrics:
+
+    // Cold compilation time (i.e., when the example is compiled with a new assembler).
+    // Hot compilation time (i.e., when the example is complied with an assembler which was already used to compile the same code).
+    // Execution time.
+    // Proving time.
+    // Verification time.
+
+    println!("============================================================");
+    println!("Benchmarking Miden examples");
+    println!("============================================================");
+
+    let args = Cli::parse();
+
+    // let's read the program
+    let program_string = fs::read_to_string(format!("../examples/{}.masm", &args.example))?;
+
+    // let's read the input fils
+    let input_string = fs::read_to_string(format!("../examples/{}.inputs", &args.example))
+        .map_err(|err| format!("Failed to open input file `{}` - {}", &args.example, err))?;
+    let inputs_des: InputFile = serde_json::from_str(&input_string)
+        .map_err(|err| format!("Failed to deserialize input data - {}", err))
+        .unwrap();
+
+    let stack_init = get_stack_init(&inputs_des);
+    let advice_tape = get_advice_tape(&inputs_des);
+
+    let inputs = get_program_inputs(&stack_init, &advice_tape);
+
+    // Compilation time
+    let now = Instant::now();
+    let assembler = Assembler::new();
+    let program = assembler
+        .compile(&program_string)
+        .expect("Could not compile source");
+
+    println! {"Compilation Time (cold): {} ms", now.elapsed().as_millis()}
+
+    let now = Instant::now();
+    let _program2 = assembler
+        .compile(&program_string)
+        .expect("Could not compile source");
+
+    println! {"Compilation Time (hot): {} ms", now.elapsed().as_millis()}
+
+    // Execution time
+    let now = Instant::now();
+    let _trace = miden::execute(&program, &inputs)
+        .map_err(|err| {
+            format!(
+                "Failed to generate exection trace = {:?}, and advice_tape = {:?}",
+                err, advice_tape
+            )
+        })
+        .unwrap();
```

Comment:
I'd prefer `expect` instead of `unwrap`, and if we're not using the result we should be able to get rid of the assignment, right?

Reply:
Actually, the unwrap is fine, sorry. But we should be able to remove the unused variable assignment